src/Authenticator/ResultInterface.php

```diff
@@ -17,32 +17,32 @@
     /**
      * General Failure
      */
-    const FAILURE = 0;
+    const FAILURE = 'FAILURE';
 
     /**
      * Failure due to identity not being found.
      */
-    const FAILURE_IDENTITY_NOT_FOUND = -1;
+    const FAILURE_IDENTITY_NOT_FOUND = 'FAILURE_IDENTITY_NOT_FOUND';
 
     /**
-     * Failure due to invalid credential being supplied.
+     * Failure due to invalid credentials being supplied.
      */
-    const FAILURE_CREDENTIAL_INVALID = -2;
+    const FAILURE_CREDENTIALS_INVALID = 'FAILURE_CREDENTIALS_INVALID';
 
     /**
-     * Failure due to other circumstances.
+     * The authentication credentials were not found in the request.
      */
-    const FAILURE_OTHER = -3;
+    const FAILURE_CREDENTIALS_NOT_IN_REQUEST = 'FAILURE_CREDENTIALS_NOT_IN_REQUEST';
 
     /**
-     * The authentication credentials were not found in the request.
+     * Failure due to other circumstances.
      */
-    const FAILURE_CREDENTIALS_NOT_FOUND = -4;
+    const FAILURE_OTHER = 'FAILURE_OTHER';
```

Comment:
Either of `FAILURE` or `FAILURE_OTHER` could be dropped. Either one would suffice for any non-specific failures.

Reply:
Is that correct @burzum - then I will remove FAILURE_OTHER and replace all instances in the code base and tests with FAILURE.

Reply:
or should we keep FAILURE_OTHER... maybe better?

Reply:
I would prefer keeping `FAILURE_OTHER`, let's see what others have to say.

Reply:
Just pick one of both, I would favor FAILURE because it's shorter and has no better or more specific meaning than FAILURE_OTHER - IMO.